src/Components/Members/Form/index.jsx

```diff
@@ -18,13 +30,83 @@ const memberModal = ({ modalOpen, onClose, addMember, data, memberId, updMember
     password: ''
   });
 
+  const addMember = async (member) => {
+    const res = await fetch(`${process.env.REACT_APP_API_URL}/api/member`, {
+      method: 'POST',
+      headers: {
+        'Content-type': 'application/json'
+      },
+      body: JSON.stringify(member)
+    });
+    if (res.status === 201) {
+      const data = await res.json();
+      setMembers([...members, data]);
+      setModalMessageOpen(false);
+      handleToast(true, 'success', 'Member added!');
+      setTimeout(() => {
+        history.push('/members');
+        handleShowToast(false);
+      }, 1500);
+    } else {
+      setModalMessageOpen(false);
+      handleToast(true, 'fail', 'Member cant be added!');
+      setTimeout(() => {
+        handleShowToast(false);
+      }, 1500);
+    }
+  };
+
+  const updMember = async (id, updatedMember) => {
+    const res = await fetch(`${process.env.REACT_APP_API_URL}/api/member/${id}`, {
+      method: 'PUT',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(updatedMember)
+    });
+    if (res.status === 200) {
+      setEditMember(null);
+      setModalMessageOpen(false);
+      handleToast(true, 'success', 'Member edited!');
+      setTimeout(() => {
+        history.push('/members');
+        handleShowToast(false);
+      }, 1500);
+    } else {
+      setModalMessageOpen(false);
+      handleToast(true, 'fail', 'Member cant be edited!');
+      setTimeout(() => {
+        handleShowToast(false);
+      }, 1500);
+    }
+  };
+
   useEffect(() => {
-    if (memberId !== null) {
-      let editMember = data.filter((member) => {
-        return member._id === memberId;
-      });
-      editMember = editMember[0];
-      const newEdit = {
+    const getMember = async (id) => {
+      try {
+        const res = await fetch(`${process.env.REACT_APP_API_URL}/api/member/${id}`);
+        const { data } = await res.json();
+        setMember({
+          name: data.name,
+          lastName: data.lastName,
+          dni: data.dni,
+          phone: data.phone,
+          email: data.email,
+          city: data.city,
+          dob: data.dob,
+          zip: data.zip,
+          isActive: false,
+          membership: data.membership,
+          password: data.password
+        });
+      } catch (error) {
+        console.log(error);
+      }
+    };
+    if (memberId) {
+      getMember(memberId);
+      editMember = member;
```

Comment:
Here, instead of have to declare every attribute, you can directly do
```suggestion
      editMember = { ...member }
```
 You are setting above the attributes for the member, so you know that member has the necessary things.